storage/merkle/proof.go

```diff
@@ -76,51 +75,46 @@ func (p *Proof) validateFormat() error {
 	keyBitCount := len(p.SiblingHashes)
 	for _, sc := range p.ShortPathLengths {
 		if keyBitCount > maxStepsForProofVerification {
-			return NewMalformedProofErrorf("number of key bits (%d) exceed limit (%d)", keyBitCount, maxStepsForProofVerification)
+			return newMalformedProofErrorf("number of key bits (%d) exceed limit (%d)", keyBitCount, maxStepsForProofVerification)
 		}
-		keyBitCount += CountUint16EncodingToInt(sc)
+		keyBitCount += int(sc)
 	}
 
 	if len(p.Key)*8 != keyBitCount {
-		return NewMalformedProofErrorf("key length doesn't match the length of ShortPathLengths and SiblingHashes")
+		return newMalformedProofErrorf("key length doesn't match the length of ShortPathLengths and SiblingHashes")
 	}
 
 	// step3 - check InterimNodeTypes
 
 	// size checks
 	if len(p.InterimNodeTypes) == 0 {
-		return NewMalformedProofErrorf("InterimNodeTypes is empty")
+		return newMalformedProofErrorf("InterimNodeTypes is empty")
 	}
 	if len(p.InterimNodeTypes) > maxKeyLength {
-		return NewMalformedProofErrorf("InterimNodeTypes is larger than max key length allowed (%d > %d)", len(p.InterimNodeTypes), maxKeyLength)
+		return newMalformedProofErrorf("InterimNodeTypes is larger than max key length allowed (%d > %d)", len(p.InterimNodeTypes), maxKeyLength)
 	}
```

Comment:
## ⚠️ 

I am not sure, but after taking a second look, I think these checks might be problematic:
* https://github.com/onflow/flow-go/blob/8f41c5c459f4be76e3afb64f4b11bbfd5b2d905d/storage/merkle/proof.go#L93-L95 `maxKeyLength` gives the key lengths in _bytes_. But theoretically, there could be a an interim node for every _bit_.
* furthermore, I think _both_ checks are redundant. We already compute precisely the expected key length: https://github.com/onflow/flow-go/blob/8f41c5c459f4be76e3afb64f4b11bbfd5b2d905d/storage/merkle/proof.go#L97-L98 Why add additional checks that are less precise and only add redundant computation? 

#### Suggestion
I think both checks above can be removed.

Reply:
> That's true, but `len(p.InterimNodeTypes)` also represents a length in bytes (the interim node type is a bit vector)

You are right. I obviously had some wires crossed in my brain. 

> I think the check `len(p.InterimNodeTypes) == 0` is redundant and is covered by the other checks.